_posts/2020-12-04-a-new-way-to-learn-and-discover-quarkus.adoc

```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title: A new way to learn & discover Quarkus (even for the Quarkus veterans)
+date: 2020-12-04
+tags: extensions codestarts quickstart
+synopsis: All our tooling has been updated and can now generate Quarkus application with example code showing the true power of the selected extensions...
+author: adamevin
+---
+
+What’s nice about this new feature is that it’s useful for fresh Quarkus developers as well as experienced ones. This is because of the wide range (and growing) of Quarkus extensions, you simply can’t know it all.
+
+Would you blame me if I say that we developers are a tiny bit lazy? We also like well documented frameworks.. But reading docs is a bit boring.. Right?
+By the way you can see that Quarkus is pretty well documented here: https://quarkus.io/guides/.
+
+We developers also love to code (at least I hope you do for your own sake :)..
+So, what would you say if, when you wish to use some new extensions, to have the code showing you their true power directly ready to run in your favorite IDE? In a few seconds?
+
+All this, using your favorite buildtool (Maven, Gradle or also Gradle with Kotlin DSL) and your favorite language (Java, Kotlin or Scala).
+
+Well, it’s been slowly making its way in for a while now (since 1.7), what we in the team call “Codestarts” are now the official way of generating your Quarkus applications starting from 1.10.2.Final!
+
+image::/assets/images/posts/codestarts/cloud-native-quarkus-app.png[align="center"]
+
+== We already prepared examples for a dozen of extensions
+
+You may spot them on https://code.quarkus.io[code.quarkus.io] (with the jet fighter icon).
+
+You can also find them on most of our Quarkus tooling:
+
+* The Quarkus Maven plugin:
++
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:create
+----
+* The Quarkus CLI, it is still experimental and don't provide an installer yet. Still, you can easily try it using the amazing https://www.jbang.dev/[JBang]
++
+[source,shell,subs=attributes+]
+----
+jbang cli@quarkusio create
+----
+* The https://quarkus.io/blog/march-of-ides/[IDE plugins] which support "Wizards w/code.quarkus.io"
+
+== A bit of fun in some of our examples
+
+My favorite one is the new Qute example with a subatomic-particles generator :)
+
+image::/assets/images/posts/codestarts/qute-and-quarks.png[align="center"]
+
+[source,shell,subs=attributes+]
+----
+jbang cli@quarkusio create qute
+----
+
+== We need more examples!
+
+Codestarts are designed to make it easy and quick to provide new examples for extensions. So whether you are an extension owner or you are keen to create an example for an extension you like, contributions are very welcome!
```

Comment:
i thought we weren't yet enabling extensions to contribute codestarts ? how is codestarts for other extensions going to work ?

Reply:
I added this note: https://github.com/quarkusio/quarkusio.github.io/pull/809/files#diff-238a39bcdaf7cd1ad761c4f1405ec6dfbeb5a8793045cf5b9cd67c617506ec36R64-R65

is that enough?